src/auth/ui/vue/show.ts

```diff
@@ -332,18 +358,232 @@ export class AuthWebview extends VueWebview {
+     */
+    startAuthFormInteraction(featureType: AuthUiElement, authType: CredentialSourceId) {
+        // Check if a previous auth interaction existed and that the new one is not the same as it
+        if (
+            this.previousFeatureType !== undefined &&
+            this.previousAuthType !== undefined &&
+            (this.previousFeatureType !== featureType || this.previousAuthType !== authType)
+        ) {
+            // At this point a user WAS previously interacting with a different auth form
+            // and started interacting with a new one (hence the new feature + auth type).
+            // We can now indicate that the previous one was cancelled and clear out any state values
+            this.endExistingAuthFormInteraction(this.previousFeatureType, this.previousAuthType)
+        }
+
+        this.previousAuthType = authType
+        this.previousFeatureType = featureType
+    }
+
+    /**
+     * The metric for when an auth form that was unsuccessfully interacted with is
+     * done being interacted with
+     */
+    private endExistingAuthFormInteraction(featureType: AuthUiElement, authType: CredentialSourceId) {
+        this.emitAuthAttempt({
+            authType,
+            featureType,
+            result: 'Cancelled',
+            invalidFields: this.invalidFields,
+            reason: 'incompleteAuthForm',
+        })
+
+        this.invalidFields = undefined
+    }
+
+    /**
+     * This is run when an auth form interaction was successful.
+     *
+     * We do this so subsequent auth form interaction checks don't
+     * assume that this was cancelled, since we look back at the following
+     * properties to see if an auth form was not successfully completed.
+     */
+    private successfulAuthFormInteraction() {
+        this.previousAuthType = undefined
+        this.previousFeatureType = undefined
+    }
+
+    /**
+     * The metric when the webview is opened.
+     *
+     * Think of this as a snapshot of the state at the start.
+     */
+    async emitOpened() {
+        telemetry.auth_addConnection.emit({
+            source: this.getSource() ?? '',
+            reason: 'opened',
+            authConnectionsCount: await this.getConnectionCount(),
+            authEnabledAreas: builderCommaDelimitedString(this.getUnlockedFeatures()),
+            result: 'Cancelled',
+        })
+    }
+
+    /**
+     * The metric emitted when the webview is closed by the user.
+     */
+    async emitClosed() {
+        const previousAuthType = this.previousAuthType
+        const previousFeatureType = this.previousFeatureType
+        if (this.previousFeatureType && this.previousAuthType) {
+            // We are closing the webview, emit a final cancellation if they were
+            // interacting with a form but failed to complete it.
+            this.endExistingAuthFormInteraction(this.previousFeatureType, this.previousAuthType)
```

Comment:
Why can't we just attach `this.invalidFields` to the event later in this method?

Reply:
The event this method emits is about a specific auth form's failed interaction, so the invalid fields are tied directly to this specific auth form. The event that comes later is a general 'closed' event.